packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergEmbedWebViewActivity.java

```diff
@@ -0,0 +1,171 @@
+    public static final String ARG_TITLE = "title";
+    private static final String JAVA_SCRIPT_INTERFACE_NAME = "wpwebkit";
+
+    protected WebView mWebView;
+
+    private ProgressBar mProgressBar;
+    private AtomicBoolean mIsWebPageLoaded = new AtomicBoolean(false);
+    private final Handler mWebPageLoadedHandler = new Handler();
+    private final Runnable mWebPageLoadedRunnable = new Runnable() {
+        @Override public void run() {
+            if (!mIsWebPageLoaded.getAndSet(true)) {
+                mProgressBar.setVisibility(View.GONE);
+            }
+        }
+    };
+
+    @SuppressLint("SetJavaScriptEnabled")
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_gutenberg_embed_web_view);
+
+        setupToolbar();
+
+        mWebView = findViewById(R.id.embed_web_view);
+
+        mProgressBar = findViewById(R.id.progress_bar);
+
+        // Set settings
+        WebSettings settings = mWebView.getSettings();
+        settings.setJavaScriptEnabled(true);
+
+        // Setup WebView client
+        setupWebViewClient();
+
+        // Setup Web Chrome client
+        mWebView.setWebChromeClient(new WebChromeClient() {
+            @Override
+            public void onProgressChanged(WebView view, int progress) {
+                if (progress == 100) {
+                    mWebPageLoadedHandler.removeCallbacks(mWebPageLoadedRunnable);
+                    mWebPageLoadedHandler.postDelayed(mWebPageLoadedRunnable, 1500);
+                } else {
+                    mIsWebPageLoaded.compareAndSet(true, false);
+                    if (mProgressBar.getVisibility() == View.GONE) {
+                        mProgressBar.setVisibility(View.VISIBLE);
+                    }
+                    mProgressBar.setProgress(progress);
+                }
+            }
+        });
+
+        load();
+    }
+
+    protected void load() {
+        mWebView.loadUrl("https://wordpress.org/gutenberg/");
+    }
+
+    private void setupToolbar() {
+        setTitle("");
+
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        if (toolbar != null) {
+            setSupportActionBar(toolbar);
+
+            ActionBar actionBar = getSupportActionBar();
+            if (actionBar != null) {
+                actionBar.setDisplayShowTitleEnabled(true);
+                actionBar.setDisplayHomeAsUpEnabled(true);
+                actionBar.setHomeAsUpIndicator(R.drawable.ic_close_24px);
+                actionBar.setSubtitle("");
+                actionBar.setTitle(getToolbarTitle());
+            }
+        }
+    }
+
+    protected String getToolbarTitle() {
+        return "";
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        super.onCreateOptionsMenu(menu);
+
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.menu_gutenberg_embed_webview, menu);
```

Comment:
Based on [this file](https://github.com/WordPress/gutenberg/pull/50440/files#diff-2379c2b642a5f24991ad4562c8edf1c023515034d5f7c76043c585a61ad2bc20), seems the menu is empty so I'm wondering if we can avoid creating it here and remove the file.

Reply:
I'll verify again but I believe we lose the back button in the WebView if this is removed.

Reply:
fixed in https://github.com/WordPress/gutenberg/pull/50440/commits/78c3d691d61ce786fcfc561518cc5e1131638dcc